psm_utils/io/ionbot.py

```diff
@@ -70,68 +74,115 @@ def __init__(
-            run=psm_dict["spectrum_file"],
-            is_decoy=(
-                True
-                if psm_dict["database"] == "D"
-                else False if psm_dict["database"] == "T" else None
-            ),
-            score=float(psm_dict["psm_score"]),
-            precursor_mz=float(psm_dict["m/z"]),
-            retention_time=float(psm_dict["observed_retention_time"]),
-            protein_list=psm_dict["proteins"].split("||"),
-            source="ionbot",
-            qvalue=float(psm_dict["q-value"]),
-            pep=float(psm_dict["PEP"]),
-            provenance_data=({"ionbot_filename": str(self.filename)}),
-            metadata={
-                col: str(psm_dict[col]) for col in psm_dict.keys() if col not in REQUIRED_COLUMNS
-            },
-        )
+    def _get_peptide_spectrum_match(self, psm_dict: dict[str, str]) -> PSM:
+        """Convert a dictionary row from ionbot CSV to a PSM object."""
+        try:
+            return PSM(
+                peptidoform=self._parse_peptidoform(
+                    psm_dict["matched_peptide"],
+                    psm_dict["modifications"],
+                    psm_dict["charge"],
+                ),
+                spectrum_id=psm_dict["spectrum_title"],
+                run=psm_dict["spectrum_file"],
+                is_decoy=(
+                    True
+                    if psm_dict["database"] == "D"
+                    else False
+                    if psm_dict["database"] == "T"
+                    else None
+                ),
+                score=float(psm_dict["psm_score"]),
+                precursor_mz=float(psm_dict["m/z"]),
+                retention_time=float(psm_dict["observed_retention_time"]),
+                protein_list=psm_dict["proteins"].split("||"),
+                source="ionbot",
+                qvalue=float(psm_dict["q-value"]),
+                pep=float(psm_dict["PEP"]),
+                provenance_data={"ionbot_filename": str(self.filename)},
+                metadata={
+                    col: str(psm_dict[col])
+                    for col in psm_dict.keys()
+                    if col not in REQUIRED_COLUMNS
+                },
+            )
+        except KeyError as e:
+            raise PSMUtilsIOException(f"Missing required column in ionbot file: {e}") from e
+        except ValueError as e:
+            raise PSMUtilsIOException(f"Error parsing numeric value in ionbot file: {e}") from e
 
     @staticmethod
-    def _parse_peptidoform(
-        peptide: str, modifications: str, charge: Union[str, int]
-    ) -> Peptidoform:
+    def _parse_peptidoform(peptide: str, modifications: str, charge: str | int) -> Peptidoform:
         """Parse peptide, modifications, and charge to Peptidoform."""
         # Split peptide into list of amino acids with termini
-        peptide = peptide = [""] + list(peptide) + [""]
+        peptide_list: list[str] = [""] + list(peptide) + [""]
```

Comment:
A bit of bikeshedding: I didn't really expect a `peptide_list` to be a list with the amino acids of a peptide, but more like a list of peptides.